### PR TITLE
[CORE-1639] Provide extensive docs for Custom Sockets usage

### PR DESCRIPTION
**Pull request by opalczynski**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -40,6 +40,11 @@ After when you want to override this setting use --instance-name in particular c
 
     syncano sync pull --instance-name new-instance-1234
 
+Documentation
```
Comment (opalczynski) [inline README.rst:43]: @lifcio native needed :)

File: docs/README.md [hunk 2]
```diff
@@ -1,123 +1,17 @@
-# How does the data transfer from Parse to Syncano work?
+# Syncano CLI documentation
```
Comment (opalczynski) [inline docs/README.md:1]: @lifcio native needed

File: docs/custom_sockets/docs.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Syncano Custom Sockets
```
Comment (opalczynski) [inline docs/custom_sockets/docs.md:1]: @lifcio native needed.

File: docs/custom_sockets/examples/advanced.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
+# Adanced Custom Socket example
```
Comment (opalczynski) [inline docs/custom_sockets/examples/advanced.md]: @lifcio native needed.

File: docs/custom_sockets/examples/hello_world.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
+# HelloWorld example
```
Comment (opalczynski) [inline docs/custom_sockets/examples/hello_world.md:1]: @lifcio native needed.

File: docs/migrations/docs.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
+# How does the data transfer from Parse to Syncano work?
```
Comment (opalczynski) [inline docs/migrations/docs.md:1]: @lifcio This is only moved - so no native is needed.

File: README.rst (hunk 1 above)
Comment (MariuszWisniewski) [reply README.rst:43]: Done

File: docs/migrations/docs.md (hunk 6 above)
Comment (MariuszWisniewski) [reply docs/migrations/docs.md:1]: K

File: docs/README.md (hunk 2 above)
Comment (MariuszWisniewski) [reply docs/README.md:1]: Done

File: docs/custom_sockets/examples/advanced.md (hunk 4 above)
Comment (MariuszWisniewski) [reply docs/custom_sockets/examples/advanced.md]: Done

File: docs/custom_sockets/examples/hello_world.md (hunk 5 above)
Comment (MariuszWisniewski) [reply docs/custom_sockets/examples/hello_world.md:1]: Done

**Comment by MariuszWisniewski** [comment]
Green light from us :)

**Pull request merged**